### ISSUE TITLE
ini and cfg by softlist title

### ISSUE THREAD

**Issue by rb6502** [mamedev/mame#808]
We have nvram and cheats per softlist item now, but control configuration and other settings are still global to the system.  I think CFG per item is an obvious home run, but INI would also be nice.

**Comment by OzFalcon** [comment]
Couldn't agree more on the CFG per game.
I think the reason it hasn't been done has something to do with the order of cfg loading.

We already have INI on a per game basis.
Simply set your inipath and put your game.ini files in the ini directory.
inipath                   $HOME/.mame;$HOME/.mame/ini

**Comment by u-man74** [comment]
I dont need CFG files at all, as long as this inconsistence is still present: http://mametesters.org/view.php?id=6068

They are only trouble making, while creating game.ini´s for vector games, as the core video options are a important part while creating them.

**Comment by rb6502** [comment]
Your "inconsistence" (English is not your native language?) is orthogonal to the issue here.  If it were fixed, there would be no reason for you to object to this feature, so keep your hands to yourself.

**Comment by u-man74** [comment]
This is not true, i would welcome this feature, but as long as the CFG files are creating trouble with .ini files, i am not happy with it. This is not orthogonal to this issue here from my POV, because before such a feature is implemented everywhere, it should get rid of my described problem, which would in turn just become bigger and heavier to solve. You may think that i have a negative attitude, but I just see this issue here, as a justified argument to bring back the described bug from MT.

**Comment by anikom15** [comment]
This would be very useful for computer games without customizable controls/joystick support.

**Comment by rb6502** [comment]
Per-software save states would be very useful for computers/consoles as well.

**Comment by etabeta78** [comment]
I know it's not exactly what you're asking, because it's not fully
flexible, but for sake of completeness (maybe some end users have never
noticed this in the documentation) let me point out that you can already
have per-software save states in current MAME: just use the -statename
parameter and set it to

%d_cart : if you want it to give a separate sta/ subfolder named after the
software mounted in the -cart media switch
%d_cdrm : if you want it to give a separate sta/ subfolder named after the
software mounted in the -cdrm media switch
%d_cass : if you want it to give a separate sta/ subfolder named after the
software mounted in the -cass media switch
%d_flop1 : if you want it to give a separate sta/ subfolder named after the
software mounted in the -flop1 media switch

and so on... it is the same kind of syntax used by -snapname, with %d_xxx
to use a folder named after the software mounted in the xxx media switch.
this automatically works for both softlist-mounted software and
fullpath-mounted software.

    f.
